src/rgw/rgw_env.cc

```diff
@@ -51,6 +51,19 @@ const char *RGWEnv::get(const char *name, const char *def_val)
   return iter->second.c_str();
 }
 
+static std::string empty_str;
+const std::string& RGWEnv::get(const std::string& name,
+			       bool *exists) const
+{
+  const auto& iter = env_map.find(name);
+  if (iter == env_map.end()) {
+    *exists = false;
+    return empty_str;
+  }
+  *exists = true;
+  return iter->second;
+}
+
```

Comment:
How about this change?  we can avoid use `exists` to check and check the return whether is none instead directly.
```cpp
const boost::optional<std::string>& RGWEnv::get(const std::string& name) const
{
  const auto& iter = env_map.find(name);
  if (iter == env_map.end()) {
    return boost::none;
  }
  return iter->second;
}
```